src/wmtk/optimization/LineSearch.cpp

```diff
@@ -26,7 +26,7 @@ bool LineSearch::check_state() const
     bool before_pass = m_invariants.before(m_interface.tuple());
     bool after_pass = true;
     for (const PrimitiveType pt : wmtk::utils::primitive_below(top_type)) {
-        after_pass |= m_invariants.after(pt, modified_simplices(pt));
+        after_pass &= m_invariants.after(pt, modified_simplices(pt));
```

Comment:
What is this function doing? It feels wrong to me that you are calling the invariant's `before` and `after` here at the same time.

Reply:
This is a funny piece of logic I introduced. In an iterative algorithm each iteration is like a vertex smooth. So you need to satisfy before and after between each step